source/common/http/conn_manager_impl.cc

```diff
@@ -367,6 +367,10 @@ void ConnectionManagerImpl::resetAllStreams(
     stream.onResetStream(StreamResetReason::ConnectionTermination, absl::string_view());
     if (response_flag.has_value()) {
       stream.stream_info_.setResponseFlag(response_flag.value());
+      if (response_flag.value() == StreamInfo::ResponseFlag::DownstreamProtocolError) {
```

Comment:
micro optimization: you already checked that is `has_value()` above. You can use the value directly `*response_flag`  and save a branch.